Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,215 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+
+            int[] totalCalories = new int[protein.Length];
+            int[] result = new int[dietPlans.Length];
+
+            for(int i=0; i<protein.Length; i++)
+            {
+                totalCalories[i] = protein[i]* 5 + carbs[i]* 5 + fat[i]* 9;
+            }
+
+            for(int i=0; i<dietPlans.Length; i++)
+            {
+                if (dietPlans[i] == "")
+                {
+                    result[i] = 0;
+                }
+                else
+                {
+                    List<int> indexes = new List<int>();
+                    result[i] = FindResult(protein, carbs, fat, totalCalories, dietPlans[i], indexes);
+                }
+            }
+
+            return result;
+
+            throw new NotImplementedException();
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindResult(int[] protein, int[] carbs, int[] fat, int[] totalCalories, string dietPlan, List<int> indexes)
+        {
+            if(dietPlan.Length > 0)
+            {
+                switch (dietPlan[0])
+                {
+                    case 'P' :  indexes = SearchMaxIndexes(protein, indexes);

```

Comment:
Move case chars to constant so that it is readable.
where `P` will be `HIGH_PROTEIN`